### PR TITLE
Support multi-home on the frontend

### PR DESCRIPTION
**Pull request by lynguyxn**
Co-authored-by: Benson Vuong bensonvuong275@gmail.com

### Administrative Info
Monday Board ID: 853606148
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- Allow users to add a new home and switch between homes
  - Change current selected home information in settings page
- Filtered recent activity on home main screen to only show the tasks completed by the current selected home
- Filtered tasks for a specific home on task screen 
- Filtered badges for a specific home on badge screen
- Added deleting home functionality on profile page
  - To prevent errors from code that references `items[currentHome]`, users cannot delete if they only have 1 home 
- Support badge number for multi-home i.e. showing badge number only for the selected home
- NOTE: done on `refactor` backend env

### Testing
How did you confirm your changes worked? 
- Ran `expo start` and manually tested the following user actions:
  - Adding a new home in user profile screen
  - Switching to a different home
  - Changing home information in settings
  - Completing a task for a specific home and confirming the badge or recent activity does not show up for another home
  - Deleting a home
  - Ensuring badge number for a specific home matches up with the number of tasks done
  - Updating a home information from profile screen

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

Video in TSE google drive: https://drive.google.com/file/d/1D54xlJRzpag-aH8nwXpsJIVUW2ou_dQK/view?usp=sharing

### CODE REVIEW

File: src/components/AuthContainer.tsx [hunk 1]
```diff
@@ -43,9 +49,11 @@ function App(): JSX.Element | null {
 
   return (
     <NavigationContainer>
+      {console.log(`in auth: ${appState}`)}
```
Comment (wllmwu) [inline src/components/AuthContainer.tsx]: Remove log

File: src/components/Profile/SettingsContainer/index.tsx [hunk 2]
```diff
@@ -13,14 +13,14 @@ import SettingsBox from '../SettingsBox';
...
+  const { userState, setUserState, currentHome } = useContext(UserContext);
 
   // workaround to parse data that may be undefined - to change?
   const homes = userState.homes.items;
   const homesString = JSON.stringify(homes);
   const homesJSON = JSON.parse(homesString);
-  const { home } = homesJSON[0];
-  const curHomeName = homesJSON[0].home.addressLine1.split(' ')[1];
+  const { home } = homesJSON[currentHome];
+  const curHomeName = homesJSON[currentHome].home.addressLine1.split(' ')[1];
```
Comment (wllmwu) [inline src/components/Profile/SettingsContainer/index.tsx:23]: Do we still need the workaround here? It just stringifies `homes` and then parses it back into an object. I'm not sure where the potential undefined might be, maybe when it tries to destructure `const { home } = homesJSON[0]`. Would it still work if we just do `const { home } = userState.homes.items[currentHome]`?

File: src/components/Tasks/TaskCompletionModal/index.tsx [hunk 3]
```diff
@@ -8,45 +8,79 @@ import SvgContainer from '../../SvgContainer';
...
+      questionID: selectedTask.id,
+      answer: 'Y',
+    };
+
+    const questionIsHome = selectedTask.type === ResponseType.HOME;
+
+    if (questionIsHome) {
+      const result: any = await API.graphql({
+        query: createResponse,
+        variables: { input: homeResponse },
```
Comment (wllmwu) [inline src/components/Tasks/TaskCompletionModal/index.tsx]: Currently both `homeResponse` and `userResponse` are created outside of this if statement, then the if statement decides which of them to use. I think we can just create the object inside the if or else statements instead. For example,
```javascript
if (questionIsHome) {
  const result: any = await API.graphql({
    query: createResponse,
    variables: {
      input: {
        homeID: ...
        questionID: ...
        answer: 'Y'
      },
    },
  });
  ...
```

File: src/contexts/UserResponseContext.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+const initialState: UserResponseState = {
+  userResponseState: {
+    items: [
+      {
+        id: '574af3cc-68d6-4051-8be2-9a6353edaeba',
+        userID: '8f3431bd62-a301-4f0b-8e52-af30d940b615',
+        questionID: '97e9d4eb-0614-4548-961c-1fa30d5c0fca',
+        createdAt: '2021-05-18T07:04:04.878Z',
+        answer: 'Yes',
+      },
```
Comment (wllmwu) [inline src/contexts/UserResponseContext.tsx]: Remove these hard coded items, they were left over from testing and never got removed. So we should just have
```javascript
  userResponseState: {
    items: [],
  },
```

**Review by wllmwu — CHANGES_REQUESTED** [review]
Couple minor changes for me, otherwise nice work!

**Review by sharmaadvay0 — APPROVED** [review]
LGTM! Great work!

**Review by wllmwu — APPROVED** [review]
Looks LGTM to me!

**Pull request merged**